src/vecs/collection.py

```diff
@@ -667,6 +704,24 @@ def create_index(
                 "HNSW Unavailable. Upgrade your pgvector installation to > 0.5.0 to enable HNSW support"
             )
 
+        # Catch case where use submits index build args for one index type but
+        # method defines a different index type.
+        if index_arguments and (
```

Comment:
please add a case here for if `auto` is used as the index method. Lets disallow (with an exception) `index_arguments` if the `method` is `auto`

a test for that combination will also be needed